test/performance/shmem_perf_suite/bw_common.h

```diff
@@ -127,10 +137,10 @@ int static inline partner_node(perf_metrics_t my_info)
         return 0;
 
     if(my_info.cstyle == COMM_PAIRWISE) {
-        int pairs = my_info.num_pes / 2;
+        int pairs = my_info.midpt;
 
-        return (my_info.my_node < pairs ? my_info.my_node + pairs :
-                    my_info.my_node - pairs);
+        return (my_info.my_node < pairs ? (my_info.my_node + pairs) :
+            (my_info.my_node - pairs));
```

Comment:
the base model (aka everything but the target usage) assumes you have a symmetric or even number of PEs.  Even with the target usage, you launch with an even set of PE's and then reduce the number you want on the left or right side -l/-r via the command line arguments